antelope_contracts/contracts/erc20/include/erc20/erc20.hpp

```diff
@@ -118,6 +118,29 @@ class [[eosio::contract]] erc20 : public contract {
     void assertnonce(eosio::name account, uint64_t next_nonce);
     using assertnonce_action = action_wrapper<"assertnonce"_n, &erc20::assertnonce>;
 
+
+   struct  message_receiver {
+
+      enum flag : uint32_t {
+         FORCE_ATOMIC = 0x1
+      };
+
+      eosio::name     account;
+      eosio::name     handler;
+      eosio::asset    min_fee;
+      uint32_t flags;
+
+      uint64_t primary_key() const { return account.value; }
+      bool has_flag(flag f) const {
+         return (flags & f) != 0;
+      }
+
+      EOSLIB_SERIALIZE(message_receiver, (account)(handler)(min_fee)(flags));
+   };
+
+typedef eosio::multi_index<"msgreceiver"_n, message_receiver> message_receiver_table;
```

Comment:
Keep this out of the contract class otherwise the ABI generator will think this is a table of this contract.

In fact there is no need for the `message_receiver` struct to even be in the header file. This can all be moved into the implementation.

Reply:
changed